src/Cli/dotnet/commands/dotnet-run/RunCommand.cs

```diff
@@ -54,133 +75,123 @@ public int Execute()
-                    {
-                        targetCommand.SetCommandArgs(launchSettings.CommandLineArgs);
-                    }
-                }
-
+                var launchSettingsCommand = ApplyLaunchSettingsProfileToCommand(targetCommand, launchSettings);
                 // Ignore Ctrl-C for the remainder of the command's execution
                 Console.CancelKeyPress += (sender, e) => { e.Cancel = true; };
-
-                return targetCommand.Execute().ExitCode;
+                return launchSettingsCommand.Execute().ExitCode;
             }
             catch (InvalidProjectFileException e)
             {
                 throw new GracefulException(
-                    string.Format(LocalizableStrings.RunCommandSpecifiecFileIsNotAValidProject, Project),
+                    string.Format(LocalizableStrings.RunCommandSpecifiecFileIsNotAValidProject, ProjectFileFullPath),
                     e);
             }
         }
 
-        public RunCommand(string configuration,
-            string framework,
-            string runtime,
-            bool noBuild,
-            string project,
-            string launchProfile,
-            bool noLaunchProfile,
-            bool noRestore,
-            bool interactive,
-            IEnumerable<string> restoreArgs,
-            IEnumerable<string> args)
+        private ICommand ApplyLaunchSettingsProfileToCommand(ICommand targetCommand, ProjectLaunchSettingsModel? launchSettings)
         {
-            Configuration = configuration;
-            Framework = framework;
-            Runtime = runtime;
-            NoBuild = noBuild;
-            Project = project;
-            LaunchProfile = launchProfile;
-            NoLaunchProfile = noLaunchProfile;
-            Args = args;
-            RestoreArgs = restoreArgs;
-            NoRestore = noRestore;
-            Interactive = interactive;
+            if (launchSettings != null)
+            {
+                if (!string.IsNullOrEmpty(launchSettings.ApplicationUrl))
+                {
+                    targetCommand.EnvironmentVariable("ASPNETCORE_URLS", launchSettings.ApplicationUrl);
+                }
+
+                targetCommand.EnvironmentVariable("DOTNET_LAUNCH_PROFILE", launchSettings.LaunchProfileName);
+
+                foreach (var entry in launchSettings.EnvironmentVariables)
+                {
+                    string value = Environment.ExpandEnvironmentVariables(entry.Value);
+                    //NOTE: MSBuild variables are not expanded like they are in VS
+                    targetCommand.EnvironmentVariable(entry.Key, value);
+                }
+                if (string.IsNullOrEmpty(targetCommand.CommandArgs) && launchSettings.CommandLineArgs != null)
+                {
+                    targetCommand.SetCommandArgs(launchSettings.CommandLineArgs);
+                }
+            }
+            return targetCommand;
```

Comment:
This is all the same functionality [as before](https://github.com/dotnet/sdk/pull/42240/files#diff-97316332fa556b0bab8849323cc14e42a5e56d86ae95466544c7ca1fb8a6bd85L57-L77), just inside a method for easier understanding.